File: backends/kotlin-ktor/src/main/kotlin/com/authorization/usecase/notification/Dto.kt
/*
 * 通知ユースケース DTO モジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
package com.authorization.usecase.notification

/**
 * 通知一括配信 DTO です。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
data class FanOutDto(
    val title:       String,
    val message:     String,
    val messageType: Int,
    val executorId:  Long,
    val url:         String = "",
)

File: backends/ruby-rails/app/usecase/notification/dto.rb
# frozen_string_literal: true
#
# 通知ユースケースの DTO を定義するモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module UseCase
  module Notification
    # 通知一斉配信時に渡す DTO です。
    FanOutDto = Struct.new(:title, :message, :message_type, :executor_id, :url, keyword_init: true)
  end
end
